### PR TITLE
WIP: cellplots

### PR DESCRIPTION
**Pull request by koehlerson**
## 2D Case
works probably. Tested fluxes from heat example
```julia
using Ferrite, SparseArrays

grid = generate_grid(Quadrilateral, (20, 20));

dim = 2
ip = Lagrange{dim, RefCube, 1}()
qr = QuadratureRule{dim, RefCube}(2)
cellvalues = CellScalarValues(qr, ip);

dh = DofHandler(grid)
push!(dh, :u, 1)
close!(dh);

K = create_sparsity_pattern(dh);

ch = ConstraintHandler(dh);

∂Ω = union(getfaceset.((grid, ), ["left", "right", "top", "bottom"])...);

dbc = Dirichlet(:u, ∂Ω, (x, t) -> 0)
add!(ch, dbc);

close!(ch)
update!(ch, 0.0);

function doassemble(cellvalues::CellScalarValues{dim}, K::SparseMatrixCSC, dh::DofHandler) where {dim}

    n_basefuncs = getnbasefunctions(cellvalues)
    Ke = zeros(n_basefuncs, n_basefuncs)
    fe = zeros(n_basefuncs)

    f = zeros(ndofs(dh))
    assembler = start_assemble(K, f)

    @inbounds for cell in CellIterator(dh)

        fill!(Ke, 0)
        fill!(fe, 0)

        reinit!(cellvalues, cell)

        for q_point in 1:getnquadpoints(cellvalues)
            dΩ = getdetJdV(cellvalues, q_point)

            for i in 1:n_basefuncs
                v  = shape_value(cellvalues, q_point, i)
                ∇v = shape_gradient(cellvalues, q_point, i)
                fe[i] += v * dΩ
                for j in 1:n_basefuncs
                    ∇u = shape_gradient(cellvalues, q_point, j)
                    Ke[i, j] += (∇v ⋅ ∇u) * dΩ
                end
            end
        end

        assemble!(assembler, celldofs(cell), fe, Ke)
    end
    return K, f
end

K, f = doassemble(cellvalues, K, dh);

apply!(K, f, ch)
u = K \ f;
function compute_heat_fluxes(cellvalues::CellScalarValues{dim,T}, dh::DofHandler, a) where {dim,T}

    n = getnbasefunctions(cellvalues)
    cell_dofs = zeros(Int, n)
    nqp = getnquadpoints(cellvalues)

    # Allocate storage for the fluxes to store
    q = [Vec{2,T}[] for _ in 1:getncells(dh.grid)]

    for (cell_num, cell) in enumerate(CellIterator(dh))
        q_cell = q[cell_num]
        celldofs!(cell_dofs, dh, cell_num)
        aᵉ = a[cell_dofs]
        reinit!(cellvalues, cell)

        for q_point in 1:nqp
            q_qp = - function_gradient(cellvalues, q_point, aᵉ)
            push!(q_cell, q_qp)
        end
    end
    return q
end
q_gp = compute_heat_fluxes(cellvalues, dh, u);
q = zeros(Vec{2},getncells(grid))
for (el, fluxes) in enumerate(q_gp)
    for flux in fluxes
        q[el] += flux
    end
    q[el] /= length(fluxes)
end

using FerriteVis
import GLMakie

plotter = MakiePlotter(dh,u)
cellplot(plotter,norm.(q,2))
```
grafik

## 3D Case

~~Is broken due to the different approach in `transfer_solution`~~. 1D celldata has now `transfer_scalar_celldata`

```
julia> include("docs/src/ferrite-examples/plasticity.jl");

julia> u,dh,uhistory,σ,κ = solve();

julia> plotter = MakiePlotter(dh,u);

julia> figure = cellplot(plotter,σ,colormap=:vik,deformation_field=:u,deformation_scale=2.0);

julia> figure
```

grafik

## What needs to be cleared

- What kind of dispatches should be supported
- how to integrate into the ferriteviewer
- what kind of quadratureplots do we want, only scalar contour plots?

### CODE REVIEW

**Comment by termi-official** [comment]
I think for now it is sufficient to provide scalar visualization and leave the computation of the quantities to the user
.

**Comment by koehlerson** [comment]
okay, I added as in all other plots the `process` option to the plot, such that a user can pass `process=x->x[1,1]` if you still have a tensorial quantity that needs to be reduced to a scalar

File: src/utils.jl [hunk 1]
```diff
@@ -295,10 +295,11 @@ function transfer_solution(plotter::MakiePlotter{3}, u::Vector; field_idx::Int=1
 
     current_vertex_index = 1
     data = fill(0.0, num_vertices(plotter), field_dim)
-    for (cell_index, cell) in enumerate(Ferrite.getcells(plotter.dh.grid))
+    for cell in Ferrite.CellIterator(dh)
+        cell_index = cell.current_cellid.x
```
Comment (termi-official) [inline src/utils.jl:299]: Out of curiosity, why is this changed?

**Review by termi-official — APPROVED** [review]
LGTM. I think we can leave the viewer integration for another PR.

File: src/utils.jl (hunk 1 above)
Comment (koehlerson) [reply src/utils.jl:299]: now its consistent :D

Comment (koehlerson) [reply src/utils.jl:299]: but I can also change it back and change instead the 2d version to also use the iteration over the `cells` array

Comment (koehlerson) [reply src/utils.jl:299]: actually, lets change to the 2D-like version without `CellIterator` because the version without it is `AbstractGrid` agnostic
